### PR TITLE
Improve the key change menu and set a few keybindings to be debug only.

### LINKED ISSUE #3847: In-game controls configuration does not detect collisions wth minetest.conf-only or hardcoded keys
The controls configuration menu is able to (kinda) detect key collisions with keys displayed on the page and displays a warning. For instance, if I make both forwards and backwards use the W key, the warning is shown.

However, the same warning is not used when either a hardcoded key is used or a key which can only be configured with minetest.conf.
For instance, if I assign F1 to anything, no warning is shown, but this is of course the default key for HUD hiding.
Also, you can assign Esc to any action but this has no effect, probably since Esc is very deeply hardcoded into Minetest. IMO the Esc key should not be assignable to anything (at least in the GUI, minetest.conf is an entirely different beast) to keep things simple.

Revision: 0d2df732e5c5075cd8848b9a9530865430630bfe

### PR DESCRIPTION
**Pull request by red-001**
See #3846. This also partially fixes #3847. 

I have added more keys to the key change menu, fixed a bug that stop the  keybinding for `forwards` from being changed if the warning about a key already being in use was visible and added a warning if the user tried to assign the `escape` to a keybinding. I set some keybindings to only have a key set to them by default in debug builds.
menu

### CODE REVIEW

File: src/guiKeyChangeMenu.cpp [hunk 1]
```diff
@@ -396,23 +405,30 @@ void GUIKeyChangeMenu::add_key(int id, const wchar_t *button_name, const std::st
...
+	this->add_key(GUI_ID_KEY_NOCLIP_BUTTON,     wgettext("Toggle noclip"),    "keymap_noclip");
+	this->add_key(GUI_ID_KEY_MINIMAP_BUTTON,    wgettext("Toggle minimap mode"),   "keymap_minimap");
+	this->add_key(GUI_ID_KEY_CAMERA_BUTTON,     wgettext("Toggle camera mode"), "keymap_camera_mode");
+	this->add_key(GUI_ID_KEY_HUD_BUTTON,        wgettext("Toggle HUD"),       "keymap_toggle_hud");
+	this->add_key(GUI_ID_KEY_TOGGLE_CHAT,       wgettext("Toggle chat"),      "keymap_toggle_chat");
+	this->add_key(GUI_ID_KEY_TOGGLE_FOG,        wgettext("Toggle fog"),       "keymap_toggle_force_fog_off");
+	this->add_key(GUI_ID_KEY_SCREENSHOT_BUTTON, wgettext("Screenshot"),       "keymap_screenshot");
+	this->add_key(GUI_ID_KEY_RANGE_BUTTON,      wgettext("Range select"),     "keymap_rangeselect");
+	this->add_key(GUI_ID_KEY_INCREASE_RANGE, wgettext("Increase veiw range"), "keymap_increase_viewing_range_min");
+	this->add_key(GUI_ID_KEY_DECREASE_RANGE, wgettext("Decrease veiw range"), "keymap_decrease_viewing_range_min");
```
Comment (kilbith) [inline src/guiKeyChangeMenu.cpp]: Typos (`veiw`).

**Comment by ObaniGemini** [comment]
`The Debug menu might also be nice, as well as camera toggle.`
(from @C1ffisme on https://github.com/minetest/minetest/issues/3846 )

File: src/defaultsettings.cpp [hunk 2]
```diff
@@ -76,11 +76,11 @@ void set_default_settings(Settings *settings)
 	settings->setDefault("tooltip_show_delay", "400");
 
 	// Some (temporary) keys for debugging
-	settings->setDefault("keymap_print_debug_stacks", "KEY_KEY_P");
-	settings->setDefault("keymap_quicktune_prev", "KEY_HOME");
-	settings->setDefault("keymap_quicktune_next", "KEY_END");
-	settings->setDefault("keymap_quicktune_dec", "KEY_NEXT");
-	settings->setDefault("keymap_quicktune_inc", "KEY_PRIOR");
+	settings->setDefault("keymap_print_debug_stacks", "");
```
Comment (est31) [inline src/defaultsettings.cpp]: this should depend on #if debug.

Comment (red-001) [inline src/defaultsettings.cpp]: done.

**Comment by Wuzzy2** [comment]
I just noticed I created a VERY similar PR:
https://github.com/minetest/minetest/issues/4537

I like the ordering of the used actions in my PR more. Maybe you can adapt this order and apply it to your PR? Then I would be willing to close my own PR.

Besides, is the strange “If this menu screws up (…)” message still relevant? It seems to be a relic of ancient times. ;-)

File: src/defaultsettings.cpp [hunk 3]
```diff
@@ -76,11 +76,19 @@ void set_default_settings(Settings *settings)
 	settings->setDefault("tooltip_show_delay", "400");
 
 	// Some (temporary) keys for debugging
+	#ifdef NDEBUG
```
Comment (rubenwardy) [inline src/defaultsettings.cpp]: I'd prefer `DEBUG` here, and swapping the two blocks, just to keep from having an unnecessary negative. It's also more consistent with the above (`keymap_toggle_update_camera`)

Comment (rubenwardy) [inline src/defaultsettings.cpp]: also, what's the code style with indentation and preprocessor guards?

**Review by rubenwardy — COMMENTED** [review]
seems handy, will review more later

File: src/defaultsettings.cpp (hunk 3 above)
Comment (sfan5) [inline src/defaultsettings.cpp]: `DEBUG` isn't defined anywhere, if you want to have block run in release use `#ifndef NDEBUG`.

Comment (sfan5) [inline src/defaultsettings.cpp]: Also concerning code style, preprocessor statements are usually without indentation regardless of all other stuff.

Comment (rubenwardy) [inline src/defaultsettings.cpp]: Look at the block above here

Comment (rubenwardy) [inline src/defaultsettings.cpp]: Debug is used in the block above here, so needs to be changed if it never exists

**Comment by bigfoot547** [comment]
> \<@Wuzzy2\>Besides, is the strange “If this menu screws up (…)” message still relevant? It seems to be a relic of ancient times. ;-) 

More like a nostalgic blast from the past. ;)

File: src/game.cpp [hunk 4]
```diff
@@ -192,6 +198,22 @@ struct LocalFormspecHandler : public TextDest {
...
+			assert(m_client != 0);
+
+			if ((fields.find("btn_continue") != fields.end())) {
+				errorstream << "server change accepted" << std::endl;
+				g_gamecallback->changeserver();
+				return;
+			}
+
+			if (fields.find("btn_exit") != fields.end()) {
+				errorstream << "server change regected" << std::endl;
```
Comment (bigfoot547) [inline src/game.cpp]: Spelling

**Review by bigfoot547 — APPROVED** [review]
I like it buuuut....

**Comment by Wuzzy2** [comment]
Why did this PR have to die?

**Pull request closed without merging**